Result/Result.swift

```diff
@@ -31,6 +31,15 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
 		}
 	}
 	
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws, non-autoclosure version
+    	public init?(_ f: () throws -> T?) {
+        	do {
+            		guard let result = try f() else { return nil }
```

Comment:
Why not make a `Break.Cancelled` error, return that and filter it out? I don't think it makes sense to add a failable initializer to `Result`.